great_expectations/datasource/fluent/spark_dbfs_datasource.py

```diff
@@ -30,57 +26,37 @@
+
     # instance attributes
     # overridden from base `Literal['spark_filesystem']`
     type: Literal["spark_dbfs"] = "spark_dbfs"  # type: ignore[assignment] # base class has different type
 
-    @public_api
-    def add_csv_asset(
-        self,
-        name: str,
-        batching_regex: Optional[Union[re.Pattern, str]] = None,
-        glob_directive: str = "**/*",
-        header: bool = False,
-        infer_schema: bool = False,
-        order_by: Optional[SortersDefinition] = None,
-        batch_metadata: Optional[BatchMetadata] = None,
-    ) -> CSVAsset:
-        """Adds a CSV DataAsset to the present "SparkDBFSDatasource" object.
-
-        Args:
-            name: The name of the CSV asset
-            batching_regex: regex pattern that matches csv filenames that is used to label the batches
-            glob_directive: glob for selecting files in DBFS directory (defaults to `**/*`) or nested directories (e.g. `*/*/*.csv`)
-            header: boolean (default False) indicating whether or not first line of CSV file is header line
-            infer_schema: boolean (default False) instructing Spark to attempt to infer schema of CSV file heuristically
-            order_by: sorting directive via either list[Sorter] or "+/- key" syntax: +/- (a/de)scending; + default
-            batch_metadata: An arbitrary user defined dictionary with string keys which will get inherited by any
-                            batches created from the asset.
-        """
-        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
-        asset = CSVAsset(
-            name=name,
-            batching_regex=batching_regex,  # type: ignore[arg-type] # pydantic will compile regex str to Pattern
-            header=header,
-            inferSchema=infer_schema,
-            order_by=order_by_sorters,
-            batch_metadata=batch_metadata or {},
-        )
-        asset._data_connector = DBFSDataConnector.build_data_connector(
+    def _build_data_connector(
+        self, data_asset: CSVAsset, glob_directive: str = "**/*", **kwargs
+    ) -> None:
+        """Builds and attaches the `DBFSDataConnector` to the asset."""
+        if kwargs:
+            raise TypeError(
+                f"_build_data_connector() got unexpected keyword arguments {list(kwargs.keys())}"
+            )
+        data_asset._data_connector = self.data_connector_type.build_data_connector(
             datasource_name=self.name,
-            data_asset_name=name,
-            batching_regex=asset.batching_regex,
+            data_asset_name=data_asset.name,
+            batching_regex=data_asset.batching_regex,
             base_directory=self.base_directory,
             glob_directive=glob_directive,
             data_context_root_directory=self.data_context_root_directory,
-            file_path_template_map_fn=DBFSPath.convert_to_protocol_version,
+            file_path_template_map_fn=DBFSPath.convert_to_file_semantics_version,
```

Comment:
@Kilo59 Unless I am not reading `GitHub` changes correctly, the protocol version (versus the semantics one) is appropriate for `DBFS` case.  Thanks.

Reply:
Oh whoops, I totally missed that.
Good catch thank you!